core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -526,6 +537,19 @@ private[spark] class ExecutorAllocationManager(
       allocationManager.synchronized {
         stageIdToNumTasks(stageId) = numTasks
         allocationManager.onSchedulerBacklogged()
+
+        var numTasksPending = 0
```

Comment:
what if a `hadoopRDD` union a `parallelizedRDD` and then do shuffle, I think in this situation a unionRDD with some parts have localities